File: jenjin-core/src/main/java/com/binarymonks/jj/async/OneTimeTask.java
package com.binarymonks.jj.async;

public abstract class OneTimeTask implements Task {

    @Override
    public void getReady() {

    }

    @Override
    public void doWork() {

    }

    @Override
    public void tearDown() {
        doOnce();
    }

    protected abstract void doOnce();

    @Override
    public boolean isDone() {
        return true;
    }
}

File: jenjin-core/src/main/kotlin/com/binarymonks/jj/core/async/OneTimeTask.kt
package com.binarymonks.jj.core.async

abstract class OneTimeTask : Task {

    override fun getReady() {

    }

    override fun doWork() {

    }

    override fun tearDown() {
        doOnce()
    }

    protected abstract fun doOnce()

    override val isDone: Boolean
        get() = true
}
